db/migrate/20170730154117_create_account_migrations.rb

```diff
@@ -0,0 +1,11 @@
+class CreateAccountMigrations < ActiveRecord::Migration[5.1]
+  def change
+    create_table :account_migrations do |t|
+      t.integer :old_person_id, null: false
+      t.integer :new_person_id, null: false
+    end
+
+    add_foreign_key :account_migrations, :people, column: :old_person_id
+    add_foreign_key :account_migrations, :people, column: :new_person_id
+  end
```

Comment:
I think you should also add unique indexes here. Even when it should never happen (because rails validates it), we had that already in the past, somehow the message was received twice and while validating it was still unique both times, but then both saved at the same time. So I trust the database more than rails ;)